server/test/controllers/aplus.test.ts

```diff
@@ -0,0 +1,445 @@
+    const url = '/v1/aplus/courses/abc';
+    await responseTests.testBadRequest(url, cookies.adminCookie).get();
+  });
+
+  it('should respond with 401 if not logged in', async () => {
+    const url = '/v1/aplus/courses/1';
+    await responseTests.testUnauthorized(url).get();
+  });
+
+  it('should respond with 502 if A+ request fails', async () => {
+    mockedAxios.get.mockImplementationOnce(_url => {
+      throw new AxiosError();
+    });
+
+    const url = '/v1/aplus/courses/1';
+    await responseTests.testBadGateway(url, cookies.adminCookie).get();
+  });
+});
+
+describe('Test POST /v1/courses/:courseId/aplus-source - add A+ grade sources', () => {
+  const getGradeSource = (
+    sourceType: AplusGradeSourceType,
+    {
+      withModuleId = false,
+      withDifficulty = false,
+      attainmentId = addGradeSourceAttainmentId,
+    }
+  ): AplusGradeSourceData => ({
+    attainmentId: attainmentId,
+    aplusCourseId: 1,
+    sourceType: sourceType,
+    moduleId: withModuleId ? 1 : undefined,
+    difficulty: withDifficulty ? 'A' : undefined,
+  });
+
+  const getFullPoints = (): AplusGradeSourceData =>
+    getGradeSource(AplusGradeSourceType.FullPoints, {});
+
+  const getModule = (): AplusGradeSourceData =>
+    getGradeSource(AplusGradeSourceType.Module, {withModuleId: true});
+
+  const getDifficulty = (): AplusGradeSourceData =>
+    getGradeSource(AplusGradeSourceType.Difficulty, {withDifficulty: true});
+
+  const checkAplusGradeSource = async (
+    gradeSource: AplusGradeSourceData
+  ): Promise<void> => {
+    const result = await AplusGradeSource.findOne({
+      where: {
+        attainmentId: gradeSource.attainmentId,
+        aplusCourseId: gradeSource.aplusCourseId,
+        sourceType: gradeSource.sourceType,
+        moduleId: gradeSource.moduleId ?? null,
+        difficulty: gradeSource.difficulty ?? null,
+      },
+    });
+
+    expect(result).not.toBe(null);
+  };
+
+  it('should add sources (admin user)', async () => {
+    const res = await request
+      .post(`/v1/courses/${courseId}/aplus-source`)
+      .send([getFullPoints(), getModule(), getDifficulty()])
+      .set('Cookie', cookies.adminCookie)
+      .expect(HttpCode.Created);
+
+    expect(JSON.stringify(res.body)).toBe('{}');
+    await checkAplusGradeSource(getFullPoints());
+    await checkAplusGradeSource(getModule());
+    await checkAplusGradeSource(getDifficulty());
+  });
+
+  it('should add sources (teacher user)', async () => {
+    const res = await request
+      .post(`/v1/courses/${courseId}/aplus-source`)
+      .send([getFullPoints(), getModule(), getDifficulty()])
+      .set('Cookie', cookies.teacherCookie)
+      .expect(HttpCode.Created);
+
+    expect(JSON.stringify(res.body)).toBe('{}');
+    await checkAplusGradeSource(getFullPoints());
+    await checkAplusGradeSource(getModule());
+    await checkAplusGradeSource(getDifficulty());
+  });
```

Comment:
Instead you could just make it a loop `for (const cookie of testCookies)` and remove the duplicate code. There are examples in other test files